src/StateTransitionOfflineClient.ts

```diff
@@ -0,0 +1,70 @@
+import { Authenticator } from '@unicitylabs/commons/lib/api/Authenticator.js';
+import { RequestId } from '@unicitylabs/commons/lib/api/RequestId.js';
+import { SubmitCommitmentStatus } from '@unicitylabs/commons/lib/api/SubmitCommitmentResponse.js';
+import { SigningService } from '@unicitylabs/commons/lib/signing/SigningService.js';
+
+import { StateTransitionClient } from './StateTransitionClient.js';
+import { Commitment } from './transaction/Commitment.js';
+import { TransactionData } from './transaction/TransactionData.js';
+
+/**
+ * Result returned when submitting a transaction to the aggregator.
+ */
+class OfflineCommitment {
+  private readonly _brand: string = 'OfflineCommitment';
+
+  /**
+   * @param requestId       Request identifier used for submission
+   * @param transactionData Submitted transaction data
+   * @param authenticator   Signature over the payload
+   */
+  public constructor(
+    public readonly requestId: RequestId,
+    public readonly transactionData: TransactionData,
+    public readonly authenticator: Authenticator,
+  ) {}
+}
+
+/**
+ * High level client implementing the token state transition workflow.
+ */
+export class StateTransitionOfflineClient extends StateTransitionClient {
```

Comment:
I think we could rename StateTransitionClient to e.g. StateTransitionBuilder (which uses Aggregator client under the hood) and then OfflineStateTransitionBuilder name starts making sense